### PR TITLE
MRG: modify `annotate` for flexibility

### LINKED ISSUE #2472: Add lineages column in `prefetch` csv output
Maybe this could be an added option to `tax annotate` or added by default to the `prefetch` output.

### PR DESCRIPTION
**Pull request by bluegenes**
`tax annotate` does not need to require the `gather` columns that are required for `tax metagenome` and `tax genome`. 
 These got especially strict after the recent taxonomy refactor.

This PR modifies `tax annotate` so that it is _just_ looking for a column it can match to `taxonomy` assignments. This will allow annotation of `search`, `prefetch` files and any other file with a good column name.

(new) Acceptable column names, in the order `annotate` will look for them:

- `name` (in `search`, `gather` output)
- `match_name` (in `prefetch` output)
- `ident`
- `accession`

Are there any others we should add? We could also let the user specify a column name via cli input, but I don't see the need to enable it unless requested?

Closes https://github.com/sourmash-bio/sourmash/issues/2472

### CODE REVIEW

**Comment by bluegenes** [comment]
@sourmash-bio/devs ready for review!

This PR is into #2469, but I think we should keep it separate and merge into `latest` after #2469 has been merged, mostly because that PR is huge and the functionality is distinct.

File: src/sourmash/tax/__main__.py [hunk 1]
```diff
@@ -299,36 +300,68 @@ def annotate(args):
...
-            w = csv.DictWriter(out_fp, header, delimiter=',')
-            w.writeheader()
-
-            for gather_res in query_gather_results:
-                for taxres in gather_res.raw_taxresults:
-                    gr = asdict(taxres.raw)
-                    write_gr = {key: gr[key] for key in gr if key in header}
-                    write_gr['lineage'] = taxres.lineageInfo.display_lineage(truncate_empty=True)
-                    w.writerow(write_gr)
+    for n, in_csv in enumerate(input_csvs):
```
Comment (ctb) [inline src/sourmash/tax/__main__.py:324]: could you make a new issue to refactor this section? I dislike really long try/except blocks but that's not a blocker for this PR (I think it's beyond scope).

**Review by ctb — APPROVED** [review]
nice and clean! LGTM!

File: src/sourmash/tax/__main__.py (hunk 1 above)
Comment (bluegenes) [reply src/sourmash/tax/__main__.py:324]: Sure -- would moving the internal annotation code to a function solve that? Or is there a different refactor you have in mind?

The try/except is just saving me from having to check the 'force' arg for each potential ValueError.

Comment (ctb) [reply src/sourmash/tax/__main__.py:324]: file the issue and find out ;).

(I'm thinking this is a place where plugins could be potentially used.)

Comment (bluegenes) [reply src/sourmash/tax/__main__.py:324]: Hah, ok! I was just going to do the refactor here if easy, but I will file an issue instead 😆

Comment (bluegenes) [reply src/sourmash/tax/__main__.py:324]: see #2520

**Pull request merged**